src/charm.py

```diff
@@ -127,24 +164,32 @@ def _on_gunicorn_pebble_ready(self, event: ops.framework.EventBase) -> None:
 
     def _configure_workload(self, event: ops.charm.EventBase) -> None:
         """Configure the workload container."""
-        pebble_config = self._get_pebble_config(event)
+        pebble_config = self._get_gunicorn_pebble_config(event)
+        statsd_pebble_config = self._get_statsd_pebble_config(event)
         if not pebble_config:
             # Charm will be in blocked status.
             return
+        juju_conf = self._check_juju_config()
+        if juju_conf:
+            self.unit.status = BlockedStatus(str(juju_conf))
+            return {}
 
         # Ensure the ingress relation has the external hostname.
         self.ingress.update_config({"service-hostname": self.config["external_hostname"]})
 
-        container = self.unit.get_container(CONTAINER_NAME)
+        container = self.unit.get_container("gunicorn")
```

Comment:
I think this should be `gunicorn_container = self.unit.get_container("gunicorn")` for clarity. References to `container` lower down here will also need changing.